### PR TITLE
SEQNG-856 Rewrote DhsClientHttp to use http4s and circe.

### PR DESCRIPTION
**Pull request by jluhrs**

### CODE REVIEW

File: modules/seqexec/server/src/main/scala/seqexec/server/keywords/DhsClientHttp.scala [hunk 1]
```diff
@@ -8,113 +8,85 @@ import java.time.format.DateTimeFormatter
...
-      case "success" =>
-        DecodeResult.ok(TrySeq(()))
-      case "error"   =>
-        (r --\ "errors").as[List[Error]].map(
-          l => TrySeq.fail[Unit](SeqexecFailure.Unexpected(l.mkString(", "))))
-      case r         =>
-        DecodeResult.fail(s"Unknown response: $r", s.history.getOrElse(CursorHistory.empty))
+  @SuppressWarnings(Array("org.wartremover.warts.Var"))
+  private val client = {
+    val max = 2
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/keywords/DhsClientHttp.scala]: Perhaps we should 3 or even more attempts. We really don't want to stop a sequence if there is an intermittent issue on the DHS, OTOH we don't want to wait too long either

File: modules/seqexec/server/src/main/scala/seqexec/server/keywords/DhsClientHttp.scala [hunk 2]
```diff
@@ -124,19 +96,72 @@ object DhsClientHttp {
   object DhsError extends ErrorType("DHS_ERROR")
   object InternalServerError extends ErrorType("INTERNAL_SERVER_ERROR")
 
-  implicit def errorTypeDecode: DecodeJson[ErrorType] = DecodeJson[ErrorType]( c =>  c.as[String].map {
+  implicit def errorTypeDecode: Decoder[ErrorType] = Decoder.instance[ErrorType](c =>  c
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/keywords/DhsClientHttp.scala]: Any reason not to use semi derived codecs?

**Comment by cquiroz** [comment]
Fixes #538

**Comment by cquiroz** [comment]
Perhaps we should test the case of simultaneous observations

File: modules/seqexec/server/src/main/scala/seqexec/server/keywords/DhsClientHttp.scala (hunk 2 above)
Comment (jluhrs) [reply modules/seqexec/server/src/main/scala/seqexec/server/keywords/DhsClientHttp.scala]: Yes. I don't know how.

**Pull request merged**